dashboard/15-final/app/ui/invoices/pagination.tsx

```diff
@@ -0,0 +1,71 @@
+'use client';
+
+import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/24/outline';
+import { usePathname, useSearchParams } from 'next/navigation';
+import clsx from 'clsx';
+import Link from 'next/link';
+
+export default function PaginationButtons({
+  totalPages,
+  currentPage,
+}: {
+  totalPages: number;
+  currentPage: number;
+}) {
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const pageNumbers = Array.from({ length: totalPages }, (_, i) => i + 1);
+
+  const createPageUrl = (pageNumber: number) => {
+    const newSearchParams = new URLSearchParams(searchParams.toString());
+    newSearchParams.set('page', pageNumber.toString());
+    return `${pathname}?${newSearchParams.toString()}`;
+  };
+
+  const PreviousTag = currentPage === 1 ? 'p' : Link;
+  const NextTag = currentPage === totalPages ? 'p' : Link;
```

Comment:
Nit. Should we call these `PreviousPageTag` and `NextPreviousTag`? I was confused about what NextTag was when reading from the bottom.

```suggestion
  const PreviousTag = currentPage === 1 ? 'p' : Link;
  const NextTag = currentPage === totalPages ? 'p' : Link;
```